source/client/ui/story/SettingsTaskView.ts

```diff
@@ -155,4 +215,37 @@ export class SettingsTree extends Tree<ITreeNode>
 
         return root.children;
     }
-}
\ No newline at end of file
+}
+
+function copyLightProperties(sourceNode: NVNode, targetNode: NVNode) {
+    // TODO: memorize properties that are not in the target light for reference when reverting a light type change
+
+    const sourceLight: CLight = sourceNode.light;
+    const targetLight: CLight = targetNode.light;
+
+    (sourceLight as any).settingProperties
+        .forEach((sourceProp: SceneUIProperty) => {
+            targetLight.ins.properties
+                .find(targetProp => targetProp.key === sourceProp.key)
+                ?.setValue(sourceProp.value);
+        });
+
+    // Convert intensity to/from legacy light types
+    // Source: https://sbcode.net/threejs/lights/
+    const legacyLightTypes = [CAmbientLight, CDirectionalLight, CHemisphereLight, CSpotLight];
+
+    const sourceLightIsLegacy: boolean = legacyLightTypes.some(lt => sourceLight instanceof lt);
+    const targetLightIsLegacy: boolean = legacyLightTypes.some(lt => targetLight instanceof lt);
+    if (sourceLightIsLegacy && !targetLightIsLegacy) {
+        targetLight.ins.setValues({ "intensity": sourceLight.ins.intensity.value / Math.PI });
+    } else if (targetLightIsLegacy && !sourceLightIsLegacy) {
+        targetLight.ins.setValues({ "intensity": sourceLight.ins.intensity.value * Math.PI });
```

Comment:
Two questions for review:
1. does it make sense to divide/multiply intensity by PI when converting from/to legacy light types?
2. Is the list of legacy light types correct?